website/addons/dryad/routes.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+from framework.routing import Rule, json_renderer
+from website.addons.dryad import views
+
+TEMPLATE_DIR = '../addons/dryad/templates/'
+
+# Routes for JSON API
+
+api_routes = {'rules':
+    [
+        Rule(
+            [
+                '/project/<pid>/dryad/settings',
+                '/project/<pid>/node/<nid>/dryad/settings',
+            ],
+            'get',
+            views.dryad_settings,
+            json_renderer,
+        ),
+        Rule(
+            [
+                '/project/<pid>/dryad/metadata',
+                '/project/<pid>/node/<nid>/dryad/metadata',
+            ],
+            'get',
+            views.dryad_get_current_metadata,
+            json_renderer,
+        ),
+        Rule(
+            [
+                '/project/<pid>/dryad/list',
+                '/project/<pid>/node/<nid>/dryad/list',
+            ],
+            'get',
+            views.dryad_list_objects,
+            json_renderer,
+        ),
+        Rule(
+            [
+                '/project/<pid>/dryad/search',
+                '/project/<pid>/node/<nid>/dryad/search',
+            ],
+            'post',
+            views.dryad_search_objects,
+            json_renderer,
+        ),
+        Rule(
+            [
+                '/project/<pid>/dryad/validate',
+                '/project/<pid>/node/<nid>/dryad/validate',
+            ],
+            'get',
+            views.dryad_validate_doi,
+            json_renderer,
+        ),
+        Rule(
```

Comment:
This rule and the one immediately below it should be a single route that accepts a put request, to set/clear the settings:

```python
Rule(
    [
        '/project/<pid>/dryad/settings/',
        '/project/<pid>/node/<nid>/dryad/settings/'
    ],
    'put',
    views.dryad_set_config,
    json_renderer,
),
```